### PR TITLE
MDAnalysis: add v2.7.0

### PR DESCRIPTION
**Pull request by RMeli**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 1]
```diff
@@ -14,12 +14,13 @@ class PyMdanalysis(PythonPackage):
     topology formats.)"""
 
     homepage = "https://www.mdanalysis.org"
-    pypi = "MDAnalysis/MDAnalysis-2.4.2.tar.gz"
+    pypi = "MDAnalysis/MDAnalysis-0.0.0.tar.gz"
 
     maintainers("RMeli")
-
+
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]:
```suggestion

```

File: var/spack/repos/builtin/packages/py-mdanalysistests/package.py [hunk 2]
```diff
@@ -10,19 +10,21 @@ class PyMdanalysistests(PythonPackage):
     """Test suite for MDAnalysis"""
 
     homepage = "https://www.mdanalysis.org"
-    pypi = "MDAnalysisTests/MDAnalysisTests-2.4.2.tar.gz"
+    pypi = "MDAnalysisTests/MDAnalysisTests-0.0.0.tar.gz"
 
     maintainers("RMeli")
-
+
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/py-mdanalysistests/package.py]:
```suggestion

```

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 3]
```diff
@@ -14,12 +14,13 @@ class PyMdanalysis(PythonPackage):
     topology formats.)"""
 
     homepage = "https://www.mdanalysis.org"
-    pypi = "MDAnalysis/MDAnalysis-2.4.2.tar.gz"
+    pypi = "MDAnalysis/MDAnalysis-0.0.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: I like to use one of the real versions so `spack url summary` works correctly.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 4]
```diff
@@ -45,7 +46,7 @@ class PyMdanalysis(PythonPackage):
     depends_on("py-numpy@1.21.0:", when="@2.5.0:", type=("build", "run"))
     depends_on("py-numpy@1.20.0:", type=("build", "run"))
 
-    depends_on("py-biopython@1.80:", type=("build", "run"))
+    depends_on("py-biopython@1.80:", when="@:2.6.1", type=("build", "run"))
     depends_on("py-networkx@2.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: Networkx is also no longer required. 

I would create a new "historical dependencies" section and move these down there so the current deps match pyproject.toml, but this is optional.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 5]
```diff
@@ -77,6 +78,7 @@ class PyMdanalysis(PythonPackage):
     depends_on("py-parmed", when="+extra_formats @2.6.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py]: The "can't be concretized with py-numpy" bug was fixed, so I think you can uncomment gsd now if you want.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 6]
```diff
@@ -77,6 +78,7 @@ class PyMdanalysis(PythonPackage):
...
 
     # analysis (extras_require)
+    depends_on("py-biopython@1.80:", when="@2.7.0: +analysis", type=("build", "run"))
     depends_on("py-seaborn", when="+analysis", type=("build", "run"))
     depends_on("py-scikit-learn", when="+analysis", type=("build", "run"))
     depends_on("py-tidynamics@1:", when="+analysis", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py:79]: Need to add networkx

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py (hunk 4 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/py-mdanalysis/package.py]: Good catch. We are trying to remove mandatory dependencies, so I will do as you suggest to clarify things.

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py (hunk 5 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/py-mdanalysis/package.py]: Thanks for the info. I added GSD versions `3.x`. `py-chemfiles` seemed to have issues on macOS with `py-cmake@3.22`, so I added a few more recent versions and confirmed that

```bash
spack install py-mdanalysis@2.7.0 +extra_formats +analysis
```

works correctly.

File: var/spack/repos/builtin/packages/py-cmake/package.py [hunk 7]
```diff
@@ -5,6 +5,13 @@
 
 from spack.package import *
 
+pycmake_versions = {
+    "3.28.1": "0d4051d101d151d8387156c463aa45c8cd0e164f870e0ac0c8c91d3ff08528e1",
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/py-cmake/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-cmake/package.py [hunk 8]
```diff
@@ -5,6 +5,12 @@
 
 from spack.package import *
 
+pycmake_versions = {
+    "3.27.9": "d8a40eef1268c91e5b520b28fd5fe0591d750e48e44276dbfd493a14ee595c41",
+    "3.22.2": "b5bd5eeb488b13cf64ec963800f3d979eaeb90b4382861b86909df503379e219",
+    "3.21.4": "30fa5ed8a5ad66dcd263adb87f3ce3dc2d0ec0ac3958f5becff577e4b62cd065",
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmake/package.py:11]: You removed 3.18.0, you should probably deprecate it first or just keep it

File: var/spack/repos/builtin/packages/py-cmake/package.py [hunk 9]
```diff
@@ -17,24 +23,23 @@ class PyCmake(PythonPackage):
...
 
-    version("3.22.2", sha256="b5bd5eeb488b13cf64ec963800f3d979eaeb90b4382861b86909df503379e219")
-    version("3.21.4", sha256="30fa5ed8a5ad66dcd263adb87f3ce3dc2d0ec0ac3958f5becff577e4b62cd065")
-    version("3.18.0", sha256="52b98c5ee70b5fa30a8623e96482227e065292f78794eb085fdf0fecb204b79b")
+    for v, sha in pycmake_versions.items():
+        version(v, sha256=sha)
 
     depends_on("ninja", type="build")
     depends_on("py-scikit-build@0.12:", type="build")
     depends_on("py-setuptools@42:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmake/package.py:32]: Missing setuptools-scm+toml

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 10]
```diff
@@ -45,30 +46,24 @@ class PyMdanalysis(PythonPackage):
...
     depends_on("py-joblib@0.12:", type=("build", "run"))
 
     depends_on("py-scipy@1.5.0:", type=("build", "run"))
-
     depends_on("py-matplotlib@1.5.1:", type=("build", "run"))
     depends_on("py-tqdm@4.43.0:", type=("build", "run"))
     depends_on("py-threadpoolctl", type=("build", "run"))
     depends_on("py-packaging", type=("build", "run"))
     depends_on("py-fasteners", type=("build", "run"))
-    depends_on("py-gsd@1.9.3:", when="@:2.5.0", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py:61]: Missing mda-xdrlib

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py [hunk 11]
```diff
@@ -45,30 +46,24 @@ class PyMdanalysis(PythonPackage):
...
 
     # extra_format (extras_require)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mdanalysis/package.py:61]: Can you sort these in the same order as pyproject.toml? Makes it easier to review. Same below.

File: var/spack/repos/builtin/packages/py-cmake/package.py (hunk 8 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/py-cmake/package.py:11]: It wasn't my intention, this is an oversight. Thanks for spotting it!

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py (hunk 11 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/py-mdanalysis/package.py:61]: Done, sorry about that.

File: var/spack/repos/builtin/packages/py-cmake/package.py [hunk 12]
```diff
@@ -17,24 +24,24 @@ class PyCmake(PythonPackage):
...
+    depends_on("setuptools-scm+toml", when="@3.27.9:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmake/package.py]:
```suggestion
    depends_on("py-setuptools-scm+toml", when="@3.27.9:", type="build")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-mdanalysis/package.py (hunk 10 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-mdanalysis/package.py:61]: Still waiting on this.

Comment (RMeli) [reply var/spack/repos/builtin/packages/py-mdanalysis/package.py:61]: Sorry, I did the changes but forgot to push. I also re-ordered according to `pyproject.toml`.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**